### ISSUE TITLE
[NVIDIA][Bug]: Need to apply rng_state_tracker to attn_dropout in GPT modeling

### ISSUE THREAD

**Issue by Wong4j** [PaddlePaddle/PaddleNLP#7101]
### 软件环境

```Markdown
- paddlepaddle:
- paddlepaddle-gpu: develop
- paddlenlp: develop
```

### 重复问题

- [X] I have searched the existing issues

### 错误描述

1. Need to add `with seed_guard_context("local_seed"):` to attn_dropout [modeling.py#L384](https://github.com/PaddlePaddle/PaddleNLP/blob/af28006018019d1b8fd17d337764a91388cd3c65/paddlenlp/transformers/gpt/modeling.py#L384)

2. `core_attn` is defined, but not used. Need to remove redundant code. [modeling.py#L309](https://github.com/PaddlePaddle/PaddleNLP/blob/af28006018019d1b8fd17d337764a91388cd3c65/paddlenlp/transformers/gpt/modeling.py#L309)

### 稳定复现步骤 & 代码

I will submit a PR to fix it.

**Comment by Wong4j** [comment]
Fixed in [PR7102](https://github.com/PaddlePaddle/PaddleNLP/pull/7102)

**Comment by ZHUI** [comment]
Sorry , can you update your PR,  this is some conflict.

**Comment by Wong4j** [comment]
@ZHUI Fixed in [PR#7141](https://github.com/PaddlePaddle/PaddleNLP/pull/7141), close this issue.
